aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -128,6 +127,42 @@ impl<'e, E: ExecutorView> StorageAdapter<'e, E> {
             Ok((buf, buf_size))
         }
     }
+
+    fn get_any_resource_size_with_layout(
+        &self,
+        address: &AccountAddress,
+        struct_tag: &StructTag,
+        metadata: &[Metadata],
+        maybe_layout: Option<&MoveTypeLayout>,
+    ) -> PartialVMResult<Option<usize>> {
+        let resource_group = get_resource_group_member_from_metadata(struct_tag, metadata);
+        if let Some(resource_group) = resource_group {
+            let key = StateKey::resource_group(address, &resource_group);
```

Comment:
We probably can deduplicate this with method above? Some special resource group handling